pytest, split requirements to main and app, unify dockerfiles (#150)
oamg/convert2rhel: CONTRIBUTING.md

1. All python code must be python 2.6/2.7/3.6 compatible
1. The code should follow linting from pylint

## Developing locally

> Dependencies: 
> https://virtualenv.pypa.io/en/latest/installation.html

Reviewer: ```suggestion
```
I don't think mentioning this is important.

Author: ok, fixed

```bash
make install  # for centOS 8 dependencies
CENTOS_VER=7 make install  # for centOS 7 dependencies

> CENTOS_VER=6 make install  # for centOS 6 dependencies

Reviewer: I don't know about other systems but on Fedora 32 and greater this is not possible.
- python26 is available on Fedora 32 but not on 33 anymore
- even with python26 installed on F32, you can't use it with virtualenv (https://developer.fedoraproject.org/tech/languages/python/multiple-pythons.html)
```
Running virtualenv with interpreter /usr/bin/python2.6
ERROR: this script requires Python 2.7 or greater.
```

Author: if you would really wish to have the development env in py26, then you could get py2.6 from the pyenv.
I tested local development in centos6 and CENTOS_6 make install works as expected

> ```bash
> make install  # for centOS 8 dependencies
> CENTOS_VER=7 make install  # for centOS 7 dependencies
> CENTOS_VER=6 make install  # for centOS 6 dependencies
> ```
> this command will create the virtual environment and installs needed dependencies.

Reviewer: ```suggestion
The command bellow creates a virtual environment and installs needed dependencies.
```bash
make install  # for CentOS 8-like environment
CENTOS_VER=7 make install  # for CentOS 7-like environment
CENTOS_VER=6 make install  # for CentOS 6-like environment
```
```

Author: Fixed

Now you have to activate the environment:
```bash

> . .venv3/bin/activate

Reviewer: ```suggestion
. .venv3/bin/activate # for CentOS 8-like environment
. .venv2.7/bin/activate # for CentOS 7-like environment
. .venv2.6/bin/activate # for CentOS 6-like environment
```

Author: fixed

```

and you're ready to run tests with:
```bash

> pytest

Reviewer: When running pytest in the .venv2.7 and .venv3, I'm getting the following for many tests:
`E   ImportError: No module named pexpect`
and 
`E   ModuleNotFoundError: No module named 'pexpect'`

Shouldn't pexpect be added to requirements/centos{7,8}.requirements.txt? Then you could remove the pexpect installation trhough yum/dnf from the Dockerfiles.

Author: Need to think about how to make it better. From one side I would like to keep app and dev deps in a separate category. But I understand that it is going to fail for the local install case. I think I know how to address this:
extending the existing requirements (you could add there `-r requirementsfilename` and other deps)

Reviewer: I'm getting `E   ImportError: No module named rpm` in `.venv2.7` now (not in `.venv3`).

```

> 

Reviewer: ```suggestion
After that, you can deactivate the environment:
```bash
deactivate
```
```

Author: fixed

## Unit tests (inside the container)

> Our unit tests are run within containers. To first create the container images, run:

Reviewer: ```suggestion
You can run unit tests also within containers. To first create the container images, run:
```

1. All python code must be python 2.6/2.7/3.6 compatible
1. The code should follow linting from pylint

## Developing locally

> The command bellow creates a virtual environment and installs needed dependencies.
> ```bash
> # for CentOS8-like system
> make install  # for CentOS 8-like system
> # for CentOS7-like system
> # CENTOS_VER=7 make install
> # for CentOS6-like system
> #. .venv2.7/bin/activate
> # CENTOS_VER=6 make install
> #. .venv2.6/bin/activate
> ```

Reviewer: ```suggestion
The commands bellow activate a virtual environment with all the necessary dependencies installed:
```bash
# for CentOS8-like system
make install  # for CentOS 8-like system
. .venv3/bin/activate
# or for CentOS7-like system
CENTOS_VER=7 make install
. .venv2.7/bin/activate
# or for CentOS6-like system
CENTOS_VER=6 make install
. .venv2.6/bin/activate
```

Author: Now this chapter looks simplier:
```md
## Developing locally

The commands below installs a python3 virtual environment
with all the necessary dependencies installed:
```bash
make install
```

and you're ready to run tests with:
```bash
make tests-locally
```
```
